components/server/src/auth/subject-id.ts

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+type ISubjectId = {
+    kind: SubjectKind;
+    /**
+     * The value of the subject id, _without_ the prefix.
+     */
+    value: string;
+};
+export type SubjectKind = keyof typeof SubjectKindNames;
+const SubjectKindNames = {
+    user: "user",
+};
+const SubjectKindByShortName: ReadonlyMap<string, SubjectKind> = new Map(
+    Object.keys(SubjectKindNames).map((k) => {
+        return [SubjectKindNames[k as SubjectKind], k as SubjectKind];
+    }),
+);
+
+export class SubjectId implements ISubjectId {
```

Comment:
Would be great to not use a class but only the shape (and call that `SubjectId`). Seems the constructor is not used anywhere else and the serialize and parse could be static.

Reply:
We discussed and agreed on the class only, because of ease of usability.